packages/amplication-client/src/VersionControl/PendingChange.tsx

```diff
@@ -86,21 +86,40 @@ const PendingChange = ({ change, linkToOrigin = false }: Props) => {
 export default PendingChange;
 
 const getEntityLinkByChangeType = (
-  type: models.EnumPendingChangeOriginType,
-  originId: string,
-  baseUrl: string
-): string => {
-  switch (type) {
-    case models.EnumPendingChangeOriginType.Entity: {
-      return `${baseUrl}/entities/${originId}`;
-    }
-    case models.EnumPendingChangeOriginType.Block: {
-      return `${baseUrl}/topics/${originId}`;
-    }
-    default: {
-      return `${baseUrl}/settings/update`;
-    }
-
-    //todo: handle: service-connections and settings from the type
-  }
+  change: models.PendingChange
+): entityLinkAndDisplayName | undefined => {
+  const changeOriginMap = {
+    [models.EnumPendingChangeOriginType.Entity]: (
+      origin: models.PendingChangeOrigin
+    ): entityLinkAndDisplayName => ({
+      relativeUrl: `entities/${change.originId}`,
+      icon: "",
+      displayName: change.origin.displayName,
+    }),
+    [models.EnumPendingChangeOriginType.Block]: (
+      origin: models.PendingChangeOrigin
+    ) => {
+      const blockTypeMap: {
+        [key in models.EnumBlockType]?: entityLinkAndDisplayName;
+      } = {
+        [models.EnumBlockType.ServiceSettings]: {
+          relativeUrl: `settings/update`,
+          icon: "",
+          displayName: "Service Settings",
+        },
+        [models.EnumBlockType.ProjectConfigurationSettings]: {
+          relativeUrl: `settings/update`,
+          icon: "",
+          displayName: "Project Settings",
+        },
+        [models.EnumBlockType.Topic]: {
+          relativeUrl: `settings/update`,
+          icon: "",
+          displayName: change.origin.displayName,
+        },
+      };
```

Comment:
We need to define this function external to the parent function, so it will not be declared for every execution of the parent